### PR TITLE
Short-circuit if-return-else-return statements

### LINKED ISSUE #757: Short-circuit if-return-else-return statements
IDEA static analysis recommendation for mobile (J2ME - constrained) environments.

### PR DESCRIPTION
**Pull request by amcp**
Fixes #757

Signed-off-by: Alexander Patrikalakis <amcp@amazon.co.jp>

-----

Thank you for contributing to JanusGraph!

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [x] Is there an issue associated with this PR? Is it referenced in the commit message?
- [x] Does your PR body contain #xyz where xyz is the issue number you are trying to resolve?
- [x] Has your PR been rebased against the latest commit within the target branch (typically `master`)?
- [x] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] Have you written and/or updated unit tests to verify your changes?
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE.txt file, including the main LICENSE.txt file in the root of this repository?
- [ ] If applicable, have you updated the NOTICE.txt file, including the main NOTICE.txt file found in the root of this repository?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?
- [ ] If this PR is a documentation-only change, have you added a `[skip ci]`
  tag to the first line of your commit message to avoid spending CPU cycles in
  Travis CI when no code, tests, or build configuration are modified?

### Note:
Please ensure that once the PR is submitted, you check Travis CI for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: janusgraph-core/src/main/java/org/janusgraph/core/attribute/Cmp.java [hunk 1]
```diff
@@ -44,13 +44,7 @@ public boolean test(Object value, Object condition) {
...
                 return value==null;
             } else {
-                if (condition.equals(value)) {
-                    return true;
-                } else if (condition.getClass().isArray()) {
-                    return ArrayUtils.isEquals(condition, value);
-                } else {
-                    return false;
-                }
+                return condition.equals(value) || condition.getClass().isArray() && ArrayUtils.isEquals(condition, value);
```
Comment (sjudeng) [inline janusgraph-core/src/main/java/org/janusgraph/core/attribute/Cmp.java]: It's correct but might add parentheses for clarity.

File: janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/ConfigNamespace.java [hunk 2]
```diff
@@ -50,9 +50,7 @@ public boolean isUmbrella() {
      * @return
      */
     public boolean hasUmbrella() {
-        if (isUmbrella()) return true;
-        if (isRoot()) return false;
-        return getNamespace().hasUmbrella();
+        return isUmbrella() || !isRoot() && getNamespace().hasUmbrella();
```
Comment (sjudeng) [inline janusgraph-core/src/main/java/org/janusgraph/diskstorage/configuration/ConfigNamespace.java]: Again consider adding parentheses for clarity.

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/query/QueryUtil.java [hunk 3]
```diff
@@ -92,35 +93,36 @@ public static InternalRelationType getType(StandardJanusGraphTx tx, String typeN
...
-                } else return false;
-            }
+        if (isQNFLiteralOrNot(condition)) {
             return true;
-        } else return false;
+        }
+        if (!(condition instanceof And)) {
+            return false;
+        }
+        for (Condition<?> child : ((And<?>) condition).getChildren()) {
```
Comment (sjudeng) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/query/QueryUtil.java]: final

File: janusgraph-core/src/main/java/org/janusgraph/graphdb/query/QueryUtil.java [hunk 4]
```diff
@@ -92,35 +93,36 @@ public static InternalRelationType getType(StandardJanusGraphTx tx, String typeN
...
 
     private static boolean isQNFLiteralOrNot(Condition<?> condition) {
-        if (condition instanceof Not) {
-            Condition child = ((Not) condition).getChild();
-            if (!isQNFLiteral(child)) return false;
-            else if (child instanceof PredicateCondition) {
-                return !((PredicateCondition) child).getPredicate().hasNegation();
-            } else return true;
-        } else return isQNFLiteral(condition);
+        if (! (condition instanceof Not)) {
```
Comment (sjudeng) [inline janusgraph-core/src/main/java/org/janusgraph/graphdb/query/QueryUtil.java]: Extra space after `!` or is that intended style?

**Comment by amcp** [comment]
@sjudeng I fixed the issues you mentioned and took note of your comment about if statements and braces for future PR. I will update pipelines-ci to include all necessary TP tests and get that merged before running TP here.

**Pull request merged**